### PR TITLE
feat: implement version checker

### PR DESCRIPTION
**Pull request by alioguzhan**

### CODE REVIEW

File: src/DecafVersionChecker.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  background-color: #444;
+}
+`;
+
+export interface DecafVersionCheckerProps {
+  currentVersion: string;
+  onNewVersion?: (newVersion: string) => void;
+}
+
+export default function DecafVersionChecker(props: DecafVersionCheckerProps) {
```
Comment (alioguzhan) [inline src/DecafVersionChecker.tsx:86]: @vst this is where all is happening.

File: src/DecafApp.tsx [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 import ZendeskWidget from 'ZendeskWidget';
 import { DecafContext, getAuthenticatedDecafClient, Principal, PublicConfig } from './context';
 import DecafSpinner from './DecafSpinner';
 
-export type DecafAppType = {
+export interface DecafAppConfig {
+  /** version of the application */
+  currentVersion?: string;
+  /** callback when a new version is available */
+  onNewVersion?: (newVersion: string) => void;
```
Comment (vst) [inline src/DecafApp.tsx]: Burada:

```js
{
    versionOld: number;
    versionNew: number;
}
```

gonderse ya?

File: src/DecafVersionChecker.tsx (hunk 1 above)
Comment (vst) [reply src/DecafVersionChecker.tsx:86]: Nice!

File: src/DecafVersionChecker.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  const [newVersion, setNewVersion] = useState();
+  const interval = useRef<number>();
+
+  useEffect(() => {
+    if (interval.current) {
+      clearInterval(interval.current);
+    }
+    interval.current = window.setInterval(() => {
+      fetch('/version.json')
```
Comment (vst) [inline src/DecafVersionChecker.tsx]: Can you append a querystring, something like `?random=${Math.random()}` to avoid cache (just in case)?

**Review by vst — CHANGES_REQUESTED** [review]
Looks nice. I did not see how it looks like though :)

A few changes are requested.

File: src/DecafVersionChecker.tsx (hunk 3 above)
Comment (vst) [reply src/DecafVersionChecker.tsx]: Hmm... Does it go to:

1. `https://hebele.hubele.com/version.json`, or
2. `https://hebele.hubele.com/webapps/zartzurt/production/version.json`?

**Comment by alioguzhan** [comment]
> Looks nice. I did not see how it looks like though :)
> 
> A few changes are requested.

Me neither :D . I am not sure if it is going to work or not. I will create a new PR on `valuation-reports` app pointing decaf-react's this branch to test it. 

I applied the changes u requested.

File: src/DecafVersionChecker.tsx (hunk 3 above)
Comment (alioguzhan) [reply src/DecafVersionChecker.tsx]: Done

**Comment by alioguzhan** [comment]
This is ready to review, again @vst

File: src/DecafVersionChecker.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            <div className="version-modal-content">
+              <p>A new version of your app is available. Please reload the page to update to the latest version.</p>
+              <span>
+                New version: <b style={{ marginRight: 10 }}>{newVersion}</b>
+                (Current version: <b>{props.currentVersion}</b>)
+              </span>
+            </div>
+            <div className="version-modal-footer">
+              <button className="cancel-btn" onClick={() => setNewVersion(undefined)}>
+                Cancel
```
Comment (vst) [inline src/DecafVersionChecker.tsx]: Instead of `Cancel`, how about `Let me finish my work first. I will reload later.`?

**Review by vst — CHANGES_REQUESTED** [review]
One last change request :)

File: src/DecafVersionChecker.tsx (hunk 4 above)
Comment (alioguzhan) [reply src/DecafVersionChecker.tsx:148]: Makes sense. Done.

**Review by vst — APPROVED** [review]
Thanks!

**Pull request merged**